unit/app/endpoints/test_query_v2.py

```
@@ -0,0 +1,872 @@
+    # Test with mcp_headers only (no token)
+    mcp_headers = {
+        "http://localhost:3000": {"X-Custom-Header": "value1"},
+        "https://git.example.com/mcp": {"X-API-Key": "secret123"},
+    }
+    tools = get_mcp_tools(servers, token=None, mcp_headers=mcp_headers)
+    assert len(tools) == 2
+    assert tools[0]["headers"] == {"X-Custom-Header": "value1"}
+    assert tools[1]["headers"] == {"X-API-Key": "secret123"}
+
+    # Test with both token and mcp_headers (should merge)
+    tools_merged = get_mcp_tools(servers, token="abc", mcp_headers=mcp_headers)
```

Ruff (0.14.8)

Possible hardcoded password assigned to argument: "token"

(S106)